@xen-orchestra/backups/Backup.js

```diff
@@ -196,34 +196,44 @@ exports.Backup = class Backup {
       ...settings[schedule.id],
     }
 
+    const vmIds = extractIdsFromSimplePattern(job.vms)
     await using(
+      Disposable.all(
+        vmIds.map(id =>
+          this._getRecord('VM', id).catch(error => {
+            runTask(
+              {
+                name: 'get VM record',
+                data: { type: 'VM', id },
+              },
+              () => Promise.reject(error)
+            )
+          })
+        )
+      ),
```

Comment:
What happens if we can't fetch a VM?

Reply:
An error will be logged. Do you mean by this question to filter `undefined` values?

Reply:
Everything is fine now.